### PR TITLE
Configurable mask branch

### PR DESCRIPTION
**Pull request by keineahnung2345**
This PR makes MASK_SHAPE configurable, and the structure of mask branch will be adjusted accordingly. 
It also adds assertion for easier debugging, ref: https://github.com/matterport/Mask_RCNN/issues/331.

One can now set MASK_SHAPE as [28,28], [56,56], [112, 112], [224, 224], etc, given MASK_POOL_SIZE is 14.

### CODE REVIEW

File: mrcnn/model.py [hunk 1]
```diff
@@ -999,8 +1000,10 @@ def build_fpn_mask_graph(rois, feature_maps, image_meta,
                            name='mrcnn_mask_bn4')(x, training=train_bn)
     x = KL.Activation('relu')(x)
 
-    x = KL.TimeDistributed(KL.Conv2DTranspose(256, (2, 2), strides=2, activation="relu"),
-                           name="mrcnn_mask_deconv")(x)
+    for i in range(num_deconv_layers):
+        x = KL.TimeDistributed(KL.Conv2DTranspose(256, (2, 2), strides=2, activation="relu"),
+                           name="mrcnn_mask_deconv" + str(i+1))(x)
```
Comment (waleedka) [inline mrcnn/model.py]: The first transposed conv should have the same name as before. In other words, `mrcnn_mask_deconv` rather than `mrcnn_mask_deconv1`. Otherwise, it will prevent loading the pre-trained COCO weights.

Comment (waleedka) [inline mrcnn/model.py]: The layer name should stay the same for the first layer. In other words, `mrcnn_mask_deconv` rather than `mrcnn_mask_deconv1`. This allows loading the pre-trained COCO weights when `num_deconv_layers` is 1.

File: mrcnn/model.py [hunk 2]
```diff
@@ -999,8 +1000,10 @@ def build_fpn_mask_graph(rois, feature_maps, image_meta,
                            name='mrcnn_mask_bn4')(x, training=train_bn)
     x = KL.Activation('relu')(x)
 
-    x = KL.TimeDistributed(KL.Conv2DTranspose(256, (2, 2), strides=2, activation="relu"),
-                           name="mrcnn_mask_deconv")(x)
+    for i in range(num_deconv_layers):
```
Comment (waleedka) [inline mrcnn/model.py]: How about adding a comment explaining that the official paper only has one transposed conv, and that additional layers is an extra feature not described in the paper?

File: mrcnn/model.py (hunk 1 above)
Comment (keineahnung2345) [reply mrcnn/model.py]: I've change the first deconv layer's name to "mrcnn_mask_deconv".

Comment (keineahnung2345) [reply mrcnn/model.py]: I've change the first deconv layer's name to "mrcnn_mask_deconv".

File: mrcnn/model.py (hunk 2 above)
Comment (keineahnung2345) [reply mrcnn/model.py]: comment added right after the explanation of `num_deconv_layers`.
